java/src/test/java/com/xxAMIDOxx/xxSTACKSxx/api/v1/menu/impl/MenuControllerImplTest.java

```diff
@@ -0,0 +1,135 @@
+package com.xxAMIDOxx.xxSTACKSxx.api.v1.menu.impl;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.xxAMIDOxx.xxSTACKSxx.api.v1.menu.SearchMenuResult;
+import com.xxAMIDOxx.xxSTACKSxx.api.v1.menu.SearchMenuResultItem;
+import com.xxAMIDOxx.xxSTACKSxx.model.Menu;
+import com.xxAMIDOxx.xxSTACKSxx.service.MenuService;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.boot.test.web.client.TestRestTemplate;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.test.context.TestPropertySource;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+import java.util.stream.Collectors;
+
+import static java.util.UUID.randomUUID;
+import static org.assertj.core.api.BDDAssertions.then;
+import static org.mockito.Mockito.when;
+
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+@TestPropertySource(properties = {"management.port=0"})
+public class MenuControllerImplTest {
+
+    @Value("${local.management.port}")
+    private int mgt;
+
+    @Autowired
+    private TestRestTemplate testRestTemplate;
+
+    @MockBean
+    private MenuService service;
+
+    private ObjectMapper mapper = new ObjectMapper();
+
+    private String getBaseURL(final int port) {
```

Comment:
This isn't specific to the menu controller. Should be in a helper class. Or is there a built in spring helper?

Reply:
This is changed to uses @LocalServerPort along with SpringBootTest.WebEnvironment.RANDOM_PORT

Reply:
Also this method is now moved to a helper class